packages/core/src/markdown-summary.ts

```diff
@@ -0,0 +1,387 @@
+   *
+   * @returns {string} string of summary buffer
+   */
+  stringify(): string {
+    return this._buffer
+  }
+
+  /**
+   * If the summary buffer is empty
+   *
+   * @returns {boolen} true if the buffer is empty
+   */
+  isEmptyBuffer(): boolean {
+    return this._buffer.length === 0
+  }
+
+  /**
+   * Resets the summary buffer without writing to summary file
+   *
+   * @returns {MarkdownSummary} markdown summary instance
+   */
+  emptyBuffer(): MarkdownSummary {
+    this._buffer = ''
+    return this
+  }
+
+  /**
+   * Adds raw text to the summary buffer
+   *
+   * @param {string} text content to add
+   * @param {boolean} [addEOL=false] (optional) append an EOL to the raw text (default: false)
+   *
+   * @returns {MarkdownSummary} markdown summary instance
+   */
+  addRaw(text: string, addEOL = false): MarkdownSummary {
+    this._buffer += text
+    return addEOL ? this.addEOL() : this
+  }
+
+  /**
+   * Adds the operating system-specific end-of-line marker to the buffer
+   *
+   * @returns {MarkdownSummary} markdown summary instance
+   */
+  addEOL(): MarkdownSummary {
+    return this.addRaw(EOL)
+  }
+
+  /**
+   * Adds an HTML codeblock to the summary buffer
+   *
+   * @param {string} code content to render within fenced code block
+   * @param {string} lang (optional) language to syntax highlight code
+   *
+   * @returns {MarkdownSummary} markdown summary instance
+   */
+  addCodeBlock(code: string, lang?: string): MarkdownSummary {
+    const attrs = {
+      ...(lang && {lang})
+    }
+    const element = this.wrap('pre', this.wrap('code', code), attrs)
+    return this.addRaw(element).addEOL()
+  }
+
+  /**
+   * Adds an HTML list to the summary buffer
+   *
+   * @param {string[]} items list of items to render
+   * @param {boolean} [ordered=false] (optional) if the rendered list should be ordered or not (default: false)
+   *
+   * @returns {MarkdownSummary} markdown summary instance
+   */
+  addList(items: string[], ordered = false): MarkdownSummary {
+    const tag = ordered ? 'ol' : 'ul'
+    const listItems = items.map(item => this.wrap('li', item)).join('')
+    const element = this.wrap(tag, listItems)
+    return this.addRaw(element).addEOL()
+  }
+
+  /**
+   * Adds an HTML table to the summary buffer
+   *
+   * @param {SummaryTableCell[]} rows table rows
+   *
+   * @returns {MarkdownSummary} markdown summary instance
+   */
+  addTable(rows: SummaryTableRow[]): MarkdownSummary {
+    const tableBody = rows
+      .map(row => {
+        const cells = row
+          .map(cell => {
+            if (typeof cell === 'string') {
+              return this.wrap('td', cell)
+            }
+
+            const {header, data, colspan, rowspan} = cell
+            const tag = header ? 'th' : 'td'
+            const attrs = {
+              ...(colspan && {colspan}),
+              ...(rowspan && {rowspan})
+            }
+
+            return this.wrap(tag, data, attrs)
+          })
+          .join('')
```

Comment:
Same here with the `.join(' ')`. You need a space if there are multiple attribute, right?

Reply:
`this.wrap` returns a tag as a string, so it'd be joining something like `['<td></td>', '<td></td>']` which doesn't need the extra space